flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/LogicalTypesTest.java

```diff
@@ -412,6 +419,40 @@ public void testNullType() {
 		assertFalse(nullType.supportsOutputConversion(int.class));
 	}
 
+	@Test
+	public void testTypeInformationAnyType() {
+		final TypeInformationAnyType<?> anyType = new TypeInformationAnyType<>(Types.TUPLE(Types.STRING, Types.INT));
+
+		testEquality(anyType, new TypeInformationAnyType<>(Types.TUPLE(Types.STRING, Types.LONG)));
+
+		testNullability(anyType);
+
+		testJavaSerializability(anyType);
+
+		testConversions(anyType, new Class[]{Tuple2.class}, new Class[]{Tuple.class});
+	}
+
+	@Test
+	public void testAnyType() {
+		testAll(
+			new AnyType<>(Human.class, new KryoSerializer<>(Human.class, new ExecutionConfig())),
+				"ANY(org.apache.flink.table.types.LogicalTypesTest$Human, " +
+					"ADNvcmcuYXBhY2hlLmZsaW5rLnRhYmxlLnR5cGVzLkxvZ2ljYWxUeXBlc1Rlc3QkSHVtYW4AAATyxpo9cAA" +
```

Comment:
Could we not hardcode the snapshot? Changes to the KryoSerializer will break this test.

Reply:
Forget this comment. It should work. We should make sure that we support old versions of internal serializer snapshots.

Reply:
I also think that hardcoding is ok in this case to indicate changes in the serialization format.